### ISSUE TITLE
ClrDataGrid clearSelection does not work when using Single Selection mode

### ISSUE THREAD

**Issue by sergiupantiru** [vmware/clarity#3106]
## Describe the bug
Selection is not cleared when applying filters if Single Section mode is used. (for Multiple selection mode the selection is cleared)

I will start with the fact that this is a bit confusing: 
- there is an issue to preserve selection when filtering (https://github.com/vmware/clarity/issues/2342)
- there is bug fix  for when selection is cleared by applying filter (https://github.com/vmware/clarity/pull/2381)

The bug fix works for the multi selection but does not fix the problem for the SingleSelection.

It looks like clearSelection only clears for MultiSelection and not for SingleSelection
https://github.com/vmware/clarity/blob/master/src/clr-angular/data/datagrid/providers/selection.ts#L149
```
public clearSelection(): void {
    this.current.length = 0;
    this.prevSelectionRefs = [];
    this.emitChange();
  }
```

I don't understand what is the expected behaviour: should the selection be cleared or should it not when applying filters.

If the selection should be cleared then it should probably also work for Single selection mode.
Probably: 
```
public clearSelection(): void {
    this.current.length = 0;
    this.prevSelectionRefs = [];
    this._currentSingle = null;
    this.prevSingleSelectionRef = null;
    this.emitChange();
  }
```
Thanks for looking into this!

**Comment by sergiupantiru** [comment]
I have the code change and unit tests. Could I make a PR and try to merge them in ?

**Comment by Jinnie** [comment]
You are welcome to do it, @sergiupantiru 
Check the development contribution guidelines if you feel uncertain about something:
https://github.com/vmware/clarity/blob/master/DEVELOPMENT_CONTRIBUTION.md

**Comment by sergiupantiru** [comment]
I have one single problem with the guidelines, the fact that i need to put my email address kind of publicly in the commit message. Is it possible to use the github version xxx@users.noreply.github.com ?

**Comment by sergiupantiru** [comment]
Can I please have a topic branch for my PR?

**Comment by mathisscott** [comment]
@sergiupantiru 
We don't make topic branches for you. Fork the repo and then you can make the topic branch off your fork. [More helpful info here](https://medium.com/claritydesignsystem/a-clean-contribution-ec9e459ce46).

**Comment by youdz** [comment]
Actually, we do make topic branches, thank you @sergiupantiru for reading our contributions guidelines entirely. 👍 

What Scott means is that for smaller PRs like this one, we don't force you to go through the hassle of a topic branch and you can just submit your single-commit PR against `master`. We put the topic branch thing in the contribution process so that we would get to pick based on the size of the contribution _before_ people submitted a PR. We've had too many 5k-lines PRs submitted against `master`, that's why we're overly careful.

TL;DR: You are good to go, just submit your PR against `master` and we'll review it directly. 😄

**Comment by sergiupantiru** [comment]
I'v created the PR. Let me know how it goes 😄

**Comment by sergiupantiru** [comment]
Hey guys, I saw that the PR is approved by 2 persons. Is there something else I need to do or should I just wait until it's merged?
